// paystack-sdk/dist/paystack.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Paystack = void 0;
const axios_1 = require("axios");
const charge_1 = require("./charge");
const customer_1 = require("./customer");
const dedicated_1 = require("./dedicated/dedicated");
const plan_1 = require("./plan");
const product_1 = require("./product/product");
const subscription_1 = require("./subscription/subscription");
const transaction_1 = require("./transaction");
const transfer_1 = require("./transfer/transfer");
/**
 * Paystack SDK
 * @author Asaju Enitan <@tPriest>
 */
class Paystack {
    constructor(key) {
        this.key = key;
        this.http = new axios_1.Axios({
            baseURL: 'https://api.paystack.co',
            headers: {
                Authorization: `Bearer ${this.key}`,
                'Content-Type': 'application/json',
            },
        });
        this.http.interceptors.response.use((response) => (response.data = JSON.parse(response.data)));
        this.charge = new charge_1.Charge(this.http);
        this.customer = new customer_1.Customer(this.http);
        this.plan = new plan_1.Plan(this.http);
        this.product = new product_1.Product(this.http);
        this.subscription = new subscription_1.Subscription(this.http);
        this.transaction = new transaction_1.Transaction(this.http);
        this.transfer = new transfer_1.Transfer(this.http);
        this.dedicated = new dedicated_1.DedicatedAccount(this.http);
    }
}
exports.Paystack = Paystack;
